File: tests/e2e/fixtures/enums/postgresql/java/src/main/java/com/example/db/com/example/db/queries/Querier.java
package com.example.db.queries;

import com.example.db.models.Priority;
import com.example.db.models.Status;
import com.example.db.models.Task;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.List;
import java.util.Optional;
import javax.sql.DataSource;

public final class Querier {
    private final DataSource dataSource;

    public Querier(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    public Optional<Task> createTask(String title, Priority priority, Status status, String description) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.createTask(conn, title, priority, status, description);
        }
    }

    public Optional<Queries.GetTaskRow> getTask(long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.getTask(conn, id);
        }
    }

    public List<Queries.ListTasksByPriorityRow> listTasksByPriority(Priority priority) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.listTasksByPriority(conn, priority);
        }
    }

    public List<Queries.ListTasksByStatusRow> listTasksByStatus(Status status) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.listTasksByStatus(conn, status);
        }
    }

    public Optional<Task> updateTaskStatus(Status status, long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.updateTaskStatus(conn, status, id);
        }
    }

    public List<Queries.ListTasksByPriorityOrAllRow> listTasksByPriorityOrAll(Priority priority) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.listTasksByPriorityOrAll(conn, priority);
        }
    }

    public List<Queries.CountByStatusRow> countByStatus() throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.countByStatus(conn);
        }
    }

    public Optional<Task> createTaskWithTags(String title, Priority priority, Status status, String description, java.util.List<Priority> tags) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.createTaskWithTags(conn, title, priority, status, description, tags);
        }
    }

    public Optional<Queries.GetTaskTagsRow> getTaskTags(long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.getTaskTags(conn, id);
        }
    }

    public Optional<Task> updateTaskTags(java.util.List<Priority> tags, long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.updateTaskTags(conn, tags, id);
        }
    }

    public void deleteTask(long id) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            Queries.deleteTask(conn, id);
        }
    }
}

File: tests/e2e/fixtures/enums/postgresql/kotlin/src/main/kotlin/com/example/db/com/example/db/queries/Querier.kt
package com.example.db.queries

import com.example.db.models.Priority
import com.example.db.models.Status
import com.example.db.models.Task
import javax.sql.DataSource

class Querier(private val dataSource: DataSource) {

    fun createTask(title: String, priority: Priority, status: Status, description: String?): Task? =
        dataSource.connection.use { conn -> Queries.createTask(conn, title, priority, status, description) }

    fun getTask(id: Long): Queries.GetTaskRow? =
        dataSource.connection.use { conn -> Queries.getTask(conn, id) }

    fun listTasksByPriority(priority: Priority): List<Queries.ListTasksByPriorityRow> =
        dataSource.connection.use { conn -> Queries.listTasksByPriority(conn, priority) }

    fun listTasksByStatus(status: Status): List<Queries.ListTasksByStatusRow> =
        dataSource.connection.use { conn -> Queries.listTasksByStatus(conn, status) }

    fun updateTaskStatus(status: Status, id: Long): Task? =
        dataSource.connection.use { conn -> Queries.updateTaskStatus(conn, status, id) }

    fun listTasksByPriorityOrAll(priority: Priority?): List<Queries.ListTasksByPriorityOrAllRow> =
        dataSource.connection.use { conn -> Queries.listTasksByPriorityOrAll(conn, priority) }

    fun countByStatus(): List<Queries.CountByStatusRow> =
        dataSource.connection.use { conn -> Queries.countByStatus(conn) }

    fun createTaskWithTags(title: String, priority: Priority, status: Status, description: String?, tags: List<Priority>): Task? =
        dataSource.connection.use { conn -> Queries.createTaskWithTags(conn, title, priority, status, description, tags) }

    fun getTaskTags(id: Long): Queries.GetTaskTagsRow? =
        dataSource.connection.use { conn -> Queries.getTaskTags(conn, id) }

    fun updateTaskTags(tags: List<Priority>, id: Long): Task? =
        dataSource.connection.use { conn -> Queries.updateTaskTags(conn, tags, id) }

    fun deleteTask(id: Long): Unit =
        dataSource.connection.use { conn -> Queries.deleteTask(conn, id) }
}
